core/src/lightclient/mod.rs

```diff
@@ -278,80 +268,56 @@ impl StarknetClient for StarknetClientImpl {
             "Cannot parse and decode the return data of Kakarot call"
         )))
     }
+    async fn storage_at(
+        &self,
+        ethereum_address: Address,
+        index: U256,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, LightClientError> {
+        // Convert the Ethereum address to a hex-encoded string
+        let address_hex = hex::encode(ethereum_address);
+        // Convert the hex-encoded string to a FieldElement
+        let ethereum_address_felt = FieldElement::from_hex_be(&address_hex).map_err(|e| {
+            LightClientError::OtherError(anyhow::anyhow!(
+                "Kakarot Core: Failed to convert Ethereum address to FieldElement: {}",
+                e
+            ))
+        })?;
+        // Prepare the calldata for the get_starknet_contract_address function call
+        let tx_calldata_vec = vec![ethereum_address_felt];
+        let request = FunctionCall {
+            contract_address: self.kakarot_account_registry,
+            entry_point_selector: GET_STARKNET_CONTRACT_ADDRESS,
+            calldata: tx_calldata_vec,
+        };
+        // Make the function call to get the Starknet contract address
+        let starknet_contract_address = self.client.call(request, &starknet_block_id).await?;
+        // Concatenate the result of the function call
+        let concatenated_result = starknet_contract_address
+            .into_iter()
+            .fold(FieldElement::ZERO, |acc, x| acc + x);
 
-    /// Get the syncing status of the light client
-    /// # Arguments
-    /// # Returns
-    ///  `Ok(SyncStatus)` if the operation was successful.
-    ///  `Err(LightClientError)` if the operation failed.
-    async fn syncing(&self) -> Result<SyncStatus, LightClientError> {
-        let status = self.client.syncing().await?;
-
-        match status {
-            SyncStatusType::NotSyncing => Ok(SyncStatus::None),
-
-            SyncStatusType::Syncing(data) => {
-                let starting_block: U256 = U256::from(data.starting_block_num);
-                let current_block: U256 = U256::from(data.current_block_num);
-                let highest_block: U256 = U256::from(data.highest_block_num);
-                let warp_chunks_amount: Option<U256> = None;
-                let warp_chunks_processed: Option<U256> = None;
+        // Prepare the calldata for the storage function call
+        // FieldElement::from(index);
+        let bytes = index.to_be_bytes();
 
-                let status_info = SyncInfo {
-                    starting_block,
-                    current_block,
-                    highest_block,
-                    warp_chunks_amount,
-                    warp_chunks_processed,
-                };
+        let f = FieldElement::from_bytes_be(&bytes).unwrap();
```

Comment:
Let's switch from using `.unwrap()` to handling if `from_bytes_be` fails, you will find other examples in the codebase where we handle from_bytes or other FieldElement methods that return a result/option using various methods (ok_or, unwrap_or_else, etc.)